presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -254,6 +260,30 @@ public void createPartitionedTableAs(HiveStorageFormat storageFormat)
         assertFalse(queryRunner.tableExists(getSession(), "test_create_partitioned_table_as"));
     }
 
+    public void createPartitionedTableAsWithIntegerColumns(HiveStorageFormat storageFormat)
+            throws Exception
+    {
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE test_create_partitioned_table_as " +
+                "WITH (" +
+                "format = '" + storageFormat + "', " +
+                "partitioned_by = ARRAY[ 'SHIP_PRIORITY', 'ORDER_STATUS' ]" +
+                ") " +
+                "AS " +
+                "SELECT orderkey AS order_key, CAST (shippriority AS INTEGER) AS ship_priority, orderstatus AS order_status " +
+                "FROM tpch.tiny.orders";
+
+        assertUpdate(createTable, "SELECT count(*) from orders");
+
+        List<HivePartition> partitions = getPartitions("test_create_partitioned_table_as");
+        assertEquals(partitions.size(), 3);
+
+        assertQuery("SELECT * from test_create_partitioned_table_as", "SELECT orderkey, CAST(shippriority AS INTEGER), orderstatus FROM orders");
+
+        assertUpdate("DROP TABLE test_create_partitioned_table_as");
```

Comment:
Can you put this in `finally` block (create table and `assertQuery` in `try`)? Just to ensure the cleanness of env after test run despite of it's outcome.